### PR TITLE
test: add test for maximum history in database settings

### PR DESCRIPTION
**Pull request by jNullj**
Add a test to the TestGui::testDatabaseSettings function for the history maximum items and maximum size settings.
The test opens the database settings dialog, disables the history items and size settings, saves the changes, reopens the dialog, and then cancels without making any changes to load the default values.
The test ensures that the default values are loaded correctly.

This tests:
1. That the two settings show and present defaults as expected.
2. Disable in gui saves the right values
3. Changing values (while the setting is disabled)
4. Gui behavior when those settings are disabled

[NOTE]: # ( Describe your changes in detail, why is this change required? )
[NOTE]: # ( Explain large or complex code modifications. )
[NOTE]: # ( If it fixes an open issue, please add "Fixes #XXX" )

## Screenshots
[TIP]:  # ( Do not include screenshots of your actual database! )
Example for affecting coverage in one of the files
gui/dbsettings/DatabaseSettingsWidgetGeneral.cpp
Before:
image
After:
image

## Testing strategy
[NOTE]: # ( Please describe in detail how you tested your changes. )
[TIP]:  # ( We expect new code to be covered by unit tests and documented with doc blocks! )
running all tests, i also added delays and run the script line by line in debug.

## Type of change
[NOTE]: # ( Please remove all lines which don't apply. )
- ❎ Bug fix (non-breaking change that fixes an issue)
- ✅ New feature (change that adds functionality)
- ❎ Breaking change (causes existing functionality to change)
- ❎ Refactor (significant modification to existing code)
- ❎ Documentation (non-code change)

### CODE REVIEW

File: tests/gui/TestGui.cpp [hunk 1]
```diff
@@ -1477,6 +1478,27 @@ void TestGui::testDatabaseSettings()
...
     QTRY_COMPARE(m_db->kdf()->rounds(), 123456);
 
+    // test disable and default values for maximum history items and size
+    triggerAction("actionDatabaseSettings");
+    auto* historyMaxItemsCheckBox = dbSettingsDialog->findChild<QCheckBox*>("historyMaxItemsCheckBox");
+    auto* historyMaxItemsSpinBox = dbSettingsDialog->findChild<QSpinBox*>("historyMaxItemsSpinBox");
+    auto* historyMaxSizeCheckBox = dbSettingsDialog->findChild<QCheckBox*>("historyMaxSizeCheckBox");
+    auto* historyMaxSizeSpinBox = dbSettingsDialog->findChild<QSpinBox*>("historyMaxSizeSpinBox");
+    // test defaults
+    QVERIFY(historyMaxItemsSpinBox->value() == Metadata::DefaultHistoryMaxItems);
```
Comment (dmaslenko) [inline tests/gui/TestGui.cpp]: To compare 2 values better to use `QCOMPARE`, it produces more detailed and clear output on fails.

File: tests/gui/TestGui.cpp [hunk 2]
```diff
@@ -1477,6 +1478,27 @@ void TestGui::testDatabaseSettings()
...
+    QVERIFY(historyMaxSizeSpinBox->value() == qRound(Metadata::DefaultHistoryMaxSize / qreal(1024 * 1024)));
+    // disable and test setting as well
+    historyMaxItemsCheckBox->setChecked(false);
+    historyMaxSizeCheckBox->setChecked(false);
+    QTest::mouseClick(dbSettingsButtonBox->button(QDialogButtonBox::Ok), Qt::LeftButton);
+    QTRY_COMPARE(m_db->metadata()->historyMaxItems(), -1);
+    QTRY_COMPARE(m_db->metadata()->historyMaxSize(), -1);
+    // then open to disabled state in gui
```
Comment (dmaslenko) [inline tests/gui/TestGui.cpp]: `// then open to check the saved disabled state in gui`
`                 ^^^       ^^^`

**Review by dmaslenko — CHANGES_REQUESTED** [review]

**Review by dmaslenko — APPROVED** [review]

**Comment by jNullj** [comment]
did a rebase to latest commit at develop

**Comment by jNullj** [comment]
did a rebase to latest commit at develop

**Comment by jNullj** [comment]
Did a rebase to latest commit at develop

**Comment by jNullj** [comment]
Did a rebase to latest commit at develop

**Comment by jNullj** [comment]
Just a rebase to latest commit

**Review by droidmonkey — APPROVED** [review]
